cli/auth.go

```diff
@@ -0,0 +1,407 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+)
+
+type authFlow struct {
+	clientID string
+	scopes   []string
+	audience string
+
+	// OpenID Discovery endpoint: see https://openid.net/specs/openid-connect-discovery-1_0.html
+	oidcDiscoveryEndpoint string
+
+	disableBrowserOpen bool
+
+	httpClient *http.Client
+	logger     golog.Logger
+	console    io.Writer
+}
+
+const (
+	defaultOpenIDDiscoveryPath = "/.well-known/openid-configuration"
+
+	prodAuthDomain = "https://auth.viam.com"
+	prodAudience   = "https://app.viam.com/"
+	prodClientID   = "HysEkkRKn6cDr2W6UFI6UYJHpiVwXFCk" // cli client
+
+	stgAuthDomain = "https://auth.viam.dev"
+	stgAudience   = "https://app.viam.dev/"
+	stgClientID   = "o75PSAO21337n6SE0IV2BF9Aj9Er9NF6" // cli client
+
+	defaultWaitInterval = time.Second * 1
+
+	//nolint:gosec
+	tokenTypeUserOAuthToken = "user-oauth-token"
```

Comment:
was it thinking we were leaking something?

Reply:
Yeah it was complaining about leaked key.